pygmt/clib/session.py

```diff
@@ -2311,3 +2312,29 @@ def extract_region(self) -> np.ndarray:
         if status != 0:
             raise GMTCLibError("Failed to extract region from current figure.")
         return region
+
+
+def raster_kind(raster: str):
+    """
+    Determine the raster kind.
+
+    >>> raster_kind("@earth_relief_01d")
+    'grid'
+    >>> raster_kind("@static_earth_relief.nc")
+    'grid'
+    >>> raster_kind("@earth_day_01d")
+    'image'
+    >>> raster_kind("@hotspots.txt")
+    """
+    with Session() as lib:
+        try:
+            img = lib.read_data(infile=raster, kind="image", mode="GMT_CONTAINER_ONLY")
+            return "image" if img.contents.header.contents.n_bands == 3 else "grid"
+        except GMTCLibError:
+            pass
+        try:
+            _ = lib.read_data(infile=raster, kind="grid", mode="GMT_CONTAINER_ONLY")
+            return "grid"
+        except GMTCLibError:
+            pass
```

Comment:
Just double-checking that there's not a function in GMT C already to determine grid/image type? I know there's [`GMT_Inquire_VirtualFile`](https://docs.generic-mapping-tools.org/6.5/devdocs/api.html#inquire-a-virtual-file-for-family) for virtualfiles, but is there not one for just files? I saw this [`GMT_Get_Info`](https://docs.generic-mapping-tools.org/6.5/devdocs/api.html#gmt-get-info), but unsure if it'll be usable here.

Reply:
https://github.com/GenericMappingTools/gmt/blob/7809736ba32d87a4a96b15444419eb176c6a35f3/src/gmt_grdio.c#L3377 

`gmt_raster_type` is the function that GMT uses to determine the raster type, but it's not a public API function.

Reply:
Oo, magic byte parsing :laughing: If that function becomes public someday, maybe we'll use it.